src/client.rs

```diff
@@ -265,19 +271,36 @@ where
         &mut self,
         games: Option<Vec<String>>,
         slots: Option<Vec<String>>,
-        tags: Option<Vec<String>>,
+        tags: Vec<String>,
         data: serde_json::Value,
     ) -> Result<(), ArchipelagoError> {
         Ok(self
             .send(ClientMessage::Bounce(Bounce {
                 games,
                 slots,
                 tags,
-                data,
+                data: BounceData::Generic(data),
             }))
             .await?)
     }
 
+    /// Sends a death link notification to the server.
+    pub async fn death_link(&mut self, options: DeathLinkOptions) -> Result<(), ArchipelagoError> {
+        self.send(ClientMessage::Bounce(Bounce {
+            games: options.games,
+            slots: options.slots,
+            tags: vec![],
+            data: BounceData::DeathLink(DeathLink {
+                time: options.time.unwrap_or_else(SystemTime::now),
+                cause: options.cause,
+                source: options
+                    .source
+                    .unwrap_or_else(|| self.room_info.seed_name.clone()),
```

Comment:
The source method in death_link_options says by default this is the slot name, but the seed is being given here